pyspeckit/spectrum/models/ammonia_radex.py

```diff
@@ -0,0 +1,638 @@
+"""
+========================================
+Ammonia inversion transition TKIN fitter
+========================================
+
+Ammonia inversion transition TKIN fitter translated from Erik Rosolowsky's
+http://svn.ok.ubc.ca/svn/signals/nh3fit/
+
+.. moduleauthor:: Adam Ginsburg <adam.g.ginsburg@gmail.com>
+
+Module API
+^^^^^^^^^^
+
+"""
+import numpy as np
+from ...mpfit import mpfit
+from ...spectrum.parinfo import ParinfoList,Parinfo
+from . import fitter
+from . import model
+import matplotlib.cbook as mpcb
+import copy
+from astropy import log
+import astropy.units as u
+from astropy import constants
+from astropy.extern.six import iteritems
+from . import mpfit_messages
+import operator
+import string
+
+from .ammonia_constants import (line_names, freq_dict, aval_dict, ortho_dict,
+                                voff_lines_dict, tau_wts_dict)
+from .ammonia_grids import ammonia_grids
+
+## Change file name!!
+grid_interpolator = ammonia_grids()
+
+TCMB = 2.7315 # K
+
+
+
+
+def ammonia_radex(xarr, tkin=20, ntot=14, width=1, xoff_v=0.0, logdens = 4.0,
+            fortho=0.0, tau=None, fillingfraction=None, return_tau=False,
+            background_tb=TCMB,
+            verbose=False, return_components=False, debug=False,
+            line_names=line_names):
```

Comment:
I think most of the contents of this function and of this module can be factored out, especially after my refactoring of the ammonia module in #136